packages/at_client/test/notification_service_test.dart

```diff
@@ -276,9 +276,35 @@ void main() {
     });
 
     test(
-        'A test to verify notification key is decrypted when shouldDecrypt is set to true',
+        'A test to verify notification value is decrypted when isEncrypted is set to null',
         () async {
-      var isEncrypted = false;
+      bool? isEncrypted;
+      var atNotification = at_notification.AtNotification(
+          '124',
+          'key-1',
+          '@alice',
+          '@bob',
+          DateTime.now().millisecondsSinceEpoch,
+          MessageTypeEnum.key.toString(),
+          isEncrypted,
+          value: 'encryptedValue');
+      var notificationResponseTransformer =
+          NotificationResponseTransformer(mockAtClientImpl);
+      notificationResponseTransformer.atKeyDecryption = mockSharedKeyDecryption;
+
+      var transformedNotification =
+          await notificationResponseTransformer.transform(Tuple()
+            ..one = atNotification
+            ..two = (NotificationConfig()
+              ..regex = '.*'
+              ..shouldDecrypt = true));
+      expect(transformedNotification.value, 'decryptedValue');
+    });
+
+    test(
+        'A test to verify notification value is decrypted when isEncrypted is set to true',
```

Comment:
We need another test, to verify that the notification value is not decrypted when (a) isEncrypted is true, and (b) shouldDecrypt is false